### ISSUE TITLE
CupertinoDatePicker: display problem on iPad

### ISSUE THREAD

**Issue by dylanalizon** [flutter/flutter#119071]
## Steps to Reproduce

1. Copy the example available on the Flutter document page for the CupertinoDatePicker component
2. Execute `flutter run` on the code sample 
3. Use iPad Simulator 
4. Open the CupertinoDatePicker 

**Actual results:** 
<img width="1348" alt="Capture d’écran 2023-01-24 à 17 09 27" src="https://user-images.githubusercontent.com/16154238/214345994-eb94c771-2a06-4f05-bf05-b4fde37f175e.png">

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/cupertino.dart';

void main() => runApp(const DatePickerApp());

class DatePickerApp extends StatelessWidget {
  const DatePickerApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const CupertinoApp(
      theme: CupertinoThemeData(brightness: Brightness.light),
      home: DatePickerExample(),
    );
  }
}

class DatePickerExample extends StatefulWidget {
  const DatePickerExample({super.key});

  @override
  State<DatePickerExample> createState() => _DatePickerExampleState();
}

class _DatePickerExampleState extends State<DatePickerExample> {
  DateTime date = DateTime(2016, 10, 26);
  DateTime time = DateTime(2016, 5, 10, 22, 35);
  DateTime dateTime = DateTime(2016, 8, 3, 17, 45);

  // This function displays a CupertinoModalPopup with a reasonable fixed height
  // which hosts CupertinoDatePicker.
  void _showDialog(Widget child) {
    showCupertinoModalPopup<void>(
        context: context,
        builder: (BuildContext context) => Container(
              height: 216,
              padding: const EdgeInsets.only(top: 6.0),
              // The Bottom margin is provided to align the popup above the system
              // navigation bar.
              margin: EdgeInsets.only(
                bottom: MediaQuery.of(context).viewInsets.bottom,
              ),
              // Provide a background color for the popup.
              color: CupertinoColors.systemBackground.resolveFrom(context),
              // Use a SafeArea widget to avoid system overlaps.
              child: SafeArea(
                top: false,
                child: child,
              ),
            ));
  }

  @override
  Widget build(BuildContext context) {
    return CupertinoPageScaffold(
      navigationBar: const CupertinoNavigationBar(
        middle: Text('CupertinoDatePicker Sample'),
      ),
      child: DefaultTextStyle(
        style: TextStyle(
          color: CupertinoColors.label.resolveFrom(context),
          fontSize: 22.0,
        ),
        child: Center(
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: <Widget>[
              _DatePickerItem(
                children: <Widget>[
                  const Text('Date'),
                  CupertinoButton(
                    // Display a CupertinoDatePicker in date picker mode.
                    onPressed: () => _showDialog(
                      CupertinoDatePicker(
                        initialDateTime: date,
                        mode: CupertinoDatePickerMode.date,
                        use24hFormat: true,
                        // This is called when the user changes the date.
                        onDateTimeChanged: (DateTime newDate) {
                          setState(() => date = newDate);
                        },
                      ),
                    ),
                    // In this example, the date is formatted manually. You can
                    // use the intl package to format the value based on the
                    // user's locale settings.
                    child: Text(
                      '${date.month}-${date.day}-${date.year}',
                      style: const TextStyle(
                        fontSize: 22.0,
                      ),
                    ),
                  ),
                ],
              ),
            ],
          ),
        ),
      ),
    );
  }
}

// This class simply decorates a row of widgets.
class _DatePickerItem extends StatelessWidget {
  const _DatePickerItem({required this.children});

  final List<Widget> children;

  @override
  Widget build(BuildContext context) {
    return DecoratedBox(
      decoration: const BoxDecoration(
        border: Border(
          top: BorderSide(
            color: CupertinoColors.inactiveGray,
            width: 0.0,
          ),
          bottom: BorderSide(
            color: CupertinoColors.inactiveGray,
            width: 0.0,
          ),
        ),
      ),
      child: Padding(
        padding: const EdgeInsets.symmetric(horizontal: 16.0),
        child: Row(
          mainAxisAlignment: MainAxisAlignment.spaceBetween,
          children: children,
        ),
      ),
    );
  }
}

```

</details>

As can be seen in the screenshot below, the days and years are not displayed correctly. You can see that it has a kind of rounding off towards the outside.

**Comment by darshankawar** [comment]
Thanks for the report @dylanalizon 
I tried your code sample using latest stable version running on iPad simulator but didn't observe the behavior you reported.

<img width="499" alt="Screenshot 2023-01-25 at 11 46 43 AM" src="https://user-images.githubusercontent.com/67046386/214495138-e1f361f6-8e31-4b7d-a871-6ccddc6046ee.png">

From the screenshot you shared, it seem to be using non-English language, I tried using `flutter_localizations` package, and added `GlobalCupertinoLocalizations` and `supportedLocales` as below to your code sample:

```
@override
  Widget build(BuildContext context) {
    return const CupertinoApp(
      localizationsDelegates: [
        GlobalCupertinoLocalizations.delegate
      ],
      supportedLocales: [
        const Locale('en'),
        const Locale('fr')
      ],
      theme: CupertinoThemeData(brightness: Brightness.light),
      home: DatePickerExample(),
    );
  }
```
But I couldn't pass `locale` to actual `CupertinoDatePicker` as the property doesn't seem to be available for that widget, so can you provide complete localized code sample or check with English language as well to see if you still see same behavior ?

Maybe your case is similar or related to https://github.com/flutter/flutter/issues/110319 that you can check and confirm.

**Comment by dylanalizon** [comment]
Thanks for the reply @darshankawar.

Indeed, I use the French locale. To put it in French you have to put only the French locale in the array supportedLocales.

I tried in English, the problem is still there but less obvious as you can see on the screenshot below.

![Capture d’écran 2023-01-25 à 09 45 21](https://user-images.githubusercontent.com/16154238/214519126-15447768-cb90-4cd9-b54c-5736dfc190d0.png)

The years on the right are not displayed correctly.

I forgot to mention that the iPad is in landscape mode. I think this is important because the width is greater than in portrait.

**Comment by dylanalizon** [comment]
Indeed on your screenshot we can see that the day is truncated. It must be related to the width. I am on an iPad Prod simulator (9.7 inch) and I don't have the problem.

**Comment by jmagman** [comment]
cc @MitchellGoodwin

**Comment by ehdwns980416** [comment]
I handled it like this because I was in a hurry.

```dart
// This function displays a CupertinoModalPopup with a reasonable fixed height
// which hosts CupertinoDatePicker.
void _showDialog(Widget child) {
  double width = MediaQuery.of(context).size.width;
  showCupertinoModalPopup<void>(
    context: context,
    builder: (BuildContext context) => Container(
      width: width,
      height: 300,
      color: CupertinoColors.systemBackground.resolveFrom(context),
      alignment: Alignment.center,
      child: Container(
        width: width > 400 ? 400 : width,
        height: 300,
        padding: const EdgeInsets.only(top: 6.0),
        // The Bottom margin is provided to align the popup above the system
        // navigation bar.
        margin: EdgeInsets.only(
          bottom: MediaQuery.of(context).viewInsets.bottom,
        ),
        // Provide a background color for the popup.
        color: CupertinoColors.systemBackground.resolveFrom(context),
        // Use a SafeArea widget to avoid system overlaps.
        child: SafeArea(
          top: false,
          child: child,
        ),
      ),
    ),
  );
}
```

![Simulator Screen Shot - iPhone 13 Pro - 2023-02-01 at 00 04 11](https://user-images.githubusercontent.com/26425092/215797067-cda3a192-4a84-4bb2-b563-d77c59fa9651.png)

**Comment by victorsanni** [comment]
@dylanalizon Does https://github.com/flutter/flutter/issues/119071 fix this issue?

**Comment by MitchellGoodwin** [comment]
Oh, yes this is fixed by #120827. Closing.
